### PR TITLE
Refactoring and preparing for a review of (more) clean architecture

### PR DESCRIPTION
**Pull request by hggutvik**
## Description
- Renaming IUnitContactPoints -> IUnitContactPointsService
- In IUserProfileService, changing the definition of _GetUserListByUuid_ to return Task<**Result<**...**>** like the other interface methods:
Related changes:
  - Moving responsibility from the service to the controller, for resolving Result.Error -> empty user profile list 
  - Updating references 
  - Updating/modifying tests that are impacted by the change
- Fix namespace of Altinn.Profile.Core/ServiceCollectionExtensions
- Other minor refactorings and/or updates, such as simplifying list initializations and correcting outdated README details

## Related Issue(s)
- #194

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Comment by hggutvik** [comment]
I ran into some network problems when trying to run the solution locally, so I haven't yet tested manually.

File: src/Altinn.Profile/Controllers/UserProfileInternalController.cs [hunk 1]
```diff
@@ -93,7 +93,11 @@ public async Task<ActionResult<List<UserProfile>>> GetList([FromBody] List<Guid>
...
         }
 
-        List<UserProfile> result = await _userProfileService.GetUserListByUuid(userUuidList);
-        return Ok(result);
+        Result<List<UserProfile>, bool> result = await _userProfileService.GetUserListByUuid(userUuidList);
+        List<UserProfile> userProfiles = result.Match(
+             userProfileList => userProfileList,
+             _ => []);
+             
+        return Ok(userProfiles);
```
Comment (SandGrainOne) [inline src/Altinn.Profile/Controllers/UserProfileInternalController.cs:101]: I must have been asleep at the wheel when we started using `Result<>` in Profile....

How did this come to your attention? Were you getting errors?

Comment (hggutvik) [reply src/Altinn.Profile/Controllers/UserProfileInternalController.cs:101]: I was trying to understand the patterns in the different classes, with a focus on the separation of concerns between Controller, (I)Service and (I)Client. That's when I noticed that `IUserProfileService.GetUserListByUuid` had a different signature than the other interface methods. 

As-is, the UserProfileService performs the resolving of the Result. My change in this PR is simply moving this step to the Controller classes

Comment (SandGrainOne) [reply src/Altinn.Profile/Controllers/UserProfileInternalController.cs:101]: I really really don't like how we're using a boolean to describe the failiure. Gives the controller nothing to work with. 

This is an example of correct use:
https://github.com/Altinn/altinn-notifications/blob/a58939e62d8b92653aa2c388b2b9a74a3ec7e501/src/Altinn.Notifications/Controllers/OrdersController.cs#L196

I'm not asking you to fix this now. Out of scope.

**Review by SandGrainOne — APPROVED** [review]

**Comment by hggutvik** [comment]
Manual testing has been performed for most functionality, but I was not able to produce the necessary test data for covering the changes related to units/contactpoints/lookup. This has to be checked in AT/TT envs

**Pull request merged**